internal/pipe/build/build.go

```diff
@@ -113,11 +113,16 @@ func doBuild(ctx *context.Context, build config.Build, target string) error {
 
 	build.Binary = binary
 	var name = build.Binary + ext
-	var path = filepath.Join(
-		ctx.Config.Dist,
-		fmt.Sprintf("%s_%s", build.ID, target),
-		name,
+	path, err := filepath.Abs(
```

Comment:
not sure why we need this either...

Reply:
To make this explanation easier, let's say this relative path here is: `dist/foo_linux_amd64/foo`.

This gets passed to the golang build builder (via `options`):
https://github.com/goreleaser/goreleaser/blob/468401bc54d3a31aebcc1526ba803760a80b51c3/internal/builders/golang/build.go#L57

which sets the output binary path to:
https://github.com/goreleaser/goreleaser/blob/468401bc54d3a31aebcc1526ba803760a80b51c3/internal/builders/golang/build.go#L113

In our example, that would be `-o dist/foo_linux_amd64/foo`.

Since the gobuild is now run in a different directory, e.g. `go/` subdirectory, this would create the `foo` binary in `go/dist/foo_linux_amd64/foo` instead of `dist/foo_linux_amd64/foo`.

And creating the release archive would fail with:
```
error=failed to add dist/foo_linux_amd64/foo -> foo to the archive: open dist/foo_linux_amd64/foo: no such file or directory
```
because that expects to find the `foo` binary in `dist/foo_linux_amd64/foo` (instead of `go/dist/foo_linux_amd64/foo`).

Hence, converting the relative path to an absolute path here seemed like a good solution.